Bug:[IPv6 only topo] [test_add_rack] test_add_rack not updated to IPv6 only topology

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

test_add_rack not updated to IPv6 only topology

### Steps to Reproduce

run tests test_add_rack on IPv6 only toppology

### Actual Behavior and Expected Behavior

The test is expected to pass on IPv6 only topology

### Relevant log output

```shell
@pytest.mark.disable_loganalyzer
    def test_add_rack(configure_dut, tbinfo, duthosts, rand_one_dut_hostname):
        global data_dir, orig_db_dir, clet_db_dir, files_dir
    
        duthost = duthosts[rand_one_dut_hostname]
    
        log_info("sys.version={}".format(sys.version))
>       do_test_add_rack(duthost, is_storage_backend='backend' in tbinfo['topo']['name'], skip_clet_test=True)

configure_dut = None
duthost    = <MultiAsicSonicHost ....>
duthosts   = [<MultiAsicSonicHost ....>]
rand_one_dut_hostname = '....'
tbinfo     = {'comment': 'Mellanox MTR testbed', 'conf-name': '....-t1-isolated-v6-d56u1-lag', 'duts': ['....'], 'duts_map': {'....': 0}, ...}

configlet/test_add_rack.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
configlet/util/base_test.py:268: in do_test_add_rack
    chk_bgp_session(duthost, tor_data["ip"]["remote"], "pre-clet test")
        duthost    = <MultiAsicSonicHost ....>
        hack_apply = False
        is_storage_backend = False
        ret        = {'changed': True, 'stdout': '', 'stderr': '', 'rc': 0, 'cmd': "python3 -c 'import generic_config_updater'", 'start': '...ne, 'removes': None, 'stdin': None}}, 'stdout_lines': [], 'stderr_lines': [], '_ansible_no_log': None, 'failed': False}
        skip_clet_test = True
        skip_generic_add = False
        skip_generic_rm = False
        skip_load  = False
        skip_prepare = False
common/configlet/utils.py:406: in chk_bgp_session
    info = duthost.get_bgp_neighbor_info(ip)
        duthost    = <MultiAsicSonicHost ....>
        ip         = ''
        msg        = 'pre-clet test'
common/devices/sonic.py:1396: in get_bgp_neighbor_info
    nbip = ipaddress.ip_address(neighbor_ip)
        asic_id    = None
        neighbor_ip = ''
        self       = <SonicHost ....>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

address = ''

    def ip_address(address):
        """Take an IP string/int and return an object of the correct type.
    
        Args:
            address: A string or integer, the IP address.  Either IPv4 or
              IPv6 addresses may be supplied; integers less than 2**32 will
              be considered to be IPv4 by default.
    
        Returns:
            An IPv4Address or IPv6Address object.
    
        Raises:
            ValueError: if the *address* passed isn't either a v4 or a v6
              address
    
        """
        try:
            return IPv4Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
        try:
            return IPv6Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
>       raise ValueError('%r does not appear to be an IPv4 or IPv6 address' %
                         address)
E       ValueError: '' does not appear to be an IPv4 or IPv6 address

address    = ''
```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/common/configlet/utils.py
+++ b/tests/common/configlet/utils.py
@@ -403,6 +403,17 @@ def chk_bgp_session(duthost, ip, msg):
         "{}: BGP session for {} = {}; expect established".format(msg, ip, bgp_state)
 
 
+def chk_any_bgp_session(duthost, msg):
+    v4_remote = tor_data.get("ip", {}).get("remote")
+    v6_remote = tor_data.get("ipv6", {}).get("remote")
+    if v4_remote:
+        chk_bgp_session(duthost, v4_remote, msg)
+    elif v6_remote:
+        chk_bgp_session(duthost, v6_remote.lower(), msg)
+    else:
+        report_error("{}: No neighbors detected".format(msg))
+
+
 def main():
     set_print()
     print("Calling compare dumps")

--- a/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
+++ b/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
@@ -628,10 +628,6 @@ configlet/test_add_rack.py:
     reason: "AddRack is not yet supported on multi-ASIC platform"
     conditions:
       - "is_multi_asic==True"
-  xfail:
-    reason: "xfail for IPv6-only topologies, need to add support for IPv6-only - https://github.com/sonic-net/sonic-mgmt/issues/20728"
-    conditions:
-      - "https://github.com/sonic-net/sonic-mgmt/issues/20728 and '-v6-' in topo_name"
 
 container_hardening/test_container_hardening.py::test_container_privileged:
   skip:

--- a/tests/configlet/util/base_test.py
+++ b/tests/configlet/util/base_test.py
@@ -11,7 +11,7 @@ from tests.common.configlet.helpers import set_log_prefix_msg, get_prefix_lvl, s
 from tests.common.configlet.utils import base_dir, data_dir, orig_db_dir, no_t0_db_dir, clet_db_dir, managed_files,\
                    patch_add_t0_dir, patch_rm_t0_dir, files_dir, tor_data, init_data, \
                    RELOAD_WAIT_TIME, PAUSE_INTF_DOWN, PAUSE_INTF_UP, PAUSE_CLET_APPLY, DB_COMP_WAIT_TIME,\
-                   do_pause, db_comp, chk_bgp_session, chk_for_pfc_wd, report_error, take_DB_dumps, init_global_data
+                   do_pause, db_comp, chk_any_bgp_session, chk_for_pfc_wd, report_error, take_DB_dumps, init_global_data
 
 
 if os.path.exists("/etc/sonic/sonic-environment"):
@@ -204,8 +204,7 @@ def apply_clet(duthost, skip_test=False):
         "DB compare failed after apply-clet"
 
     # Ensure BGP session is up
-    chk_bgp_session(duthost, tor_data["ip"]["remote"], "post-clet test")
-    chk_bgp_session(duthost, tor_data["ipv6"]["remote"].lower(), "post-clet test")
+    chk_any_bgp_session(duthost, "post-clet test")
 
     log_info("AddRack by template succeeded")
 
@@ -265,8 +264,7 @@ def do_test_add_rack(duthost, is_storage_backend=False, skip_load=False,
                      is_storage_backend=is_storage_backend)
 
         # Ensure BGP session is up before we apply stripped minigraph
-        chk_bgp_session(duthost, tor_data["ip"]["remote"], "pre-clet test")
-        chk_bgp_session(duthost, tor_data["ipv6"]["remote"].lower(), "pre-clet test")
+        chk_any_bgp_session(duthost, "pre-clet test")
 
         set_log_prefix_msg("test prepare")
         prepare_for_test(duthost)

